File: src/mVasu.Api.Contracts/TiskilistaCardDto.cs
namespace mVasu.Api.Contracts;

/// <summary>
/// Compact card view for the Tiskilista list. Field types mirror the XPO
/// schema exactly: vuokra is double, neliot is float, kerros is a free-form
/// string ("2/4"), so the JSON shape stays lossless.
/// </summary>
public sealed record TiskilistaCardDto(
    Guid Id,
    string Osoite,
    int? Kptunnus,
    int? Huonetunnus,
    string? Tyyppi,
    string? Laji,
    double? Vuokra,
    DateTimeOffset? Vapautuu,
    float? Neliot,
    string? Kerros,
    string? Kerroksia,
    string? Tila,
    string? SopimusTila,
    string? Kunta,
    string? Kaupunginosa,
    string? Prio,
    string? Isannoitsija,
    string? Markkinoija,
    bool LumoFi,
    bool Vuokraovi,
    bool OnKuvausTarve,
    bool Hissi,
    bool Parveke,
    bool Sauna,
    double? Latitude,
    double? Longitude,
    double? DistanceKm);

File: src/mVasu.Pwa/src/app/core/models/tiskilista-card.dto.ts
/**
 * Mirrors mVasu.Api.Contracts.TiskilistaCardDto. Vuokra is a number (XPO
 * stores it as double), neliot is a number (XPO single), kerros/kerroksia
 * are free-form strings ("2/4" etc.).
 */
export interface TiskilistaCardDto {
  readonly id: string;
  readonly osoite: string;
  readonly kptunnus: number | null;
  readonly huonetunnus: number | null;
  readonly tyyppi: string | null;
  readonly laji: string | null;
  readonly vuokra: number | null;
  readonly vapautuu: string | null;
  readonly neliot: number | null;
  readonly kerros: string | null;
  readonly kerroksia: string | null;
  readonly tila: string | null;
  readonly sopimusTila: string | null;
  readonly kunta: string | null;
  readonly kaupunginosa: string | null;
  readonly prio: string | null;
  readonly isannoitsija: string | null;
  readonly markkinoija: string | null;
  readonly lumoFi: boolean;
  readonly vuokraovi: boolean;
  readonly onKuvausTarve: boolean;
  readonly hissi: boolean;
  readonly parveke: boolean;
  readonly sauna: boolean;
  readonly nextEsittelyAt: string | null;
  readonly latitude: number | null;
  readonly longitude: number | null;
  readonly distanceKm: number | null;
}
